### ISSUE TITLE
[rt.openssl.org #4677] Options after parameters are ignored in OpenSSL 1.1.0

### ISSUE THREAD

**Issue by richsalz** [openssl/openssl#2421]
Migrated from [rt.openssl.org#4677](https://rt.openssl.org/Ticket/Display.html?id=4677) (status was 'open')

Requestors:
* peter@lekensteyn.nl

From peter@lekensteyn.nl on 2016-09-17 17:54:11:
```
Hi,

Commands which execute normally with OpenSSL 1.0.2h fail in OpenSSL
1.1.0. Presumably after the "Big apps cleanup (option-parsing, etc)",

Options after parameters are no longer interpreted. For example,
'openssl dhparam 128 -out /dev/null' used to discard the DH params
output, but since 1.0.2 it no longer happens.

I also noticed that 'openssl genrsa -help' no longer displays the
'numbits' option.
-- 
Kind regards,
Peter Wu
https://lekensteyn.nl

```

From levitte@openssl.org on 2016-09-17 20:18:31:
```
On Sat Sep 17 17:54:11 2016, peter@lekensteyn.nl wrote:
> Hi,
>
> Commands which execute normally with OpenSSL 1.0.2h fail in OpenSSL
> 1.1.0. Presumably after the "Big apps cleanup (option-parsing, etc)",
>
> Options after parameters are no longer interpreted. For example,
> 'openssl dhparam 128 -out /dev/null' used to discard the DH params
> output, but since 1.0.2 it no longer happens.

You're right.  There should at least be a warning

> I also noticed that 'openssl genrsa -help' no longer displays the
> 'numbits' option.

Good point.

I'll have a closer look at this and the other commands in the next few days.

Cheers,
Richard

-- 
Richard Levitte
levitte@openssl.org
```

**Comment by Lekensteyn** [comment]
Still a problem in the latest master (0e2c7b3ee), tested with the `dhparam` option.

**Comment by levitte** [comment]
I'm sure I did work on this...  not all the way to make an actual PR, it seems.  I'll have a look through my pile of branches and see if I can find it.

**Comment by richsalz** [comment]
The reporting 'numbits' see OPT_HELP_STR in others apps files.
As for intermixing parmeters and flags, NO. :)

**Comment by levitte** [comment]
Nono, there's no effort to intermix them.  But we should definitely raise a warning or an error about extra arguments that get ignored otherwise.

**Comment by richsalz** [comment]
Got it.  Many commands test for argc after parsing.  the test in dhparam (and rsagen) is wrong.

**Comment by richsalz** [comment]
fixed with #4201.
